Symfony/CS/Tests/AbstractIntegrationTest.php

```diff
@@ -0,0 +1,271 @@
+
+/**
+ * Integration test base class.
+ *
+ * This test searches for '.test' fixture files in the given directory.
+ * Each fixture file will be parsed and tested against the expected result.
+ *
+ * Fixture files have the following format:
+ *
+ * --TEST--
+ * Example test description
+ * --CONFIG--
+ * level=symfony|none|psr0|psr1|psr2|symfony
+ * fixers=fixer1,fixer2,...*
+ * --fixers=fixer3,fixer4,...**
+ * --INPUT--
+ * Code to fix
+ * --EXPECT--
+ * Expected code after fixing***
+ *
+ *   * Additional fixers may be omitted.
+ *  ** Black listed filters may be omitted.
+ * *** When the expected block is omitted the input is expected not to
+ *     be changed by the fixers.
+ *
+ * @author SpacePossum
+ *
+ * @internal
+ */
+abstract class AbstractIntegrationTest extends \PHPUnit_Framework_TestCase
+{
+    private static $builtInFixers;
+
+    public static function setUpBeforeClass()
+    {
+        $tmpFile = static::getTempFile();
+        if (!is_file($tmpFile)) {
+            $dir = dirname($tmpFile);
+            if (!is_dir($dir)) {
+                $fs = new Filesystem();
+                $fs->mkdir($dir, 0766);
+            }
+        }
+    }
+
+    public static function tearDownAfterClass()
+    {
+        unlink(static::getTempFile());
+    }
+
+    /**
+     * @dataProvider getTests
+     *
+     * @see doTestIntegration()
+     */
+    public function testIntegration($testFileName, $testTitle, $fixers, $input, $expected = null)
+    {
+        $this->doTestIntegration($testFileName, $testTitle, $fixers, $input, $expected);
+    }
+
+    /**
+     * Creates test data by parsing '.test' files.
+     *
+     * @return array
+     */
+    public function getTests()
+    {
+        $fixturesDir = realpath($this->getFixturesDir());
+        if (!is_dir($fixturesDir)) {
+            throw new \UnexpectedValueException(sprintf('Given fixture dir "%s" is not a directory.', $fixturesDir));
+        }
+
+        $tests = array();
+
+        foreach (new \RecursiveIteratorIterator(new \RecursiveDirectoryIterator($fixturesDir), \RecursiveIteratorIterator::LEAVES_ONLY) as $file) {
+            if (!preg_match('/\.test$/', $file)) {
+                continue;
+            }
+
+            $test = file_get_contents($file->getRealpath());
+            $fileName = $file->getFileName();
+            if (!preg_match('/--TEST--[\r\n](.*?)\s--CONFIG--[\r\n](.*?)\s--INPUT--[\r\n](.*?[\r\n]*)(?:[\r\n]--EXPECT--\s(.*)|$)/s', $test, $match)) {
+                throw new \InvalidArgumentException(sprintf('Test format invalid for "%s".', $fileName));
+            }
+
+            $tests[] = array($fileName, $match[1], $this->getFixersFromConfig($fileName, $match[2]), $match[3], isset($match[4]) ? $match[4] : null);
+        }
+
+        return $tests;
+    }
+
+    /**
+     * Returns the full path to directory which contains the tests.
+     *
+     * @return string
+     */
+    abstract protected static function getFixturesDir();
```

Comment:
Please, let me write a comment to remember that this method must not be both abstract and static.